### PR TITLE
Add example to play ABC files, and remove SetBuzzerFrequency command

### PR DESCRIPTION
**Pull request by qwandor**
It's a bit of a hack but it's just an example for the `messages` crate, so no need to review that part too closely.

### CODE REVIEW

File: messages/src/client.rs [hunk 1]
```diff
@@ -88,17 +90,13 @@ impl Hoverkite {
...
-        self.send_command(Side::Left, command)
-    }
-
     /// Plays the given sequence of notes on the hoverboard.
     pub fn play_notes(&mut self, notes: &[Note]) -> Result<(), io::Error> {
         for note in notes {
+            trace!("Sending {:?}", note);
             let command = Command::AddBuzzerNote(*note);
             self.send_command(Side::Left, command)?;
+            sleep(NOTE_SEND_SLEEP_DURATION);
```
Comment (alsuren) [inline messages/src/client.rs:99]: :/

can we rename this function to `play_notes_blocking()` or something suitably scary, to make it obvious that you don't want to include it in your control loop?

File: messages/examples/abc.rs [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+    let port = serialport::new(&port_name, BAUD_RATE)
+        .open()
+        .map_err(|e| error!("Failed to open serial port {}: {}", port_name, e))
+        .ok();
+
+    let mut hoverkite = Hoverkite::new(port, None);
+
+    let tune = abc_parser::abc::tune(
+        "X:1
```
Comment (alsuren) [inline messages/examples/abc.rs:43]: add attribution as a comment with a url?

File: messages/examples/abc.rs [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+use std::num::NonZeroU32;
+use std::process::exit;
+use std::thread;
+use std::time::Duration;
+
+const BAUD_RATE: u32 = 115_200;
+const SLEEP_DURATION: Duration = Duration::from_millis(2);
+
+/// Tempo in BPM.
+const TEMPO: f32 = 300.0;
```
Comment (alsuren) [inline messages/examples/abc.rs:15]: turbozurka

File: messages/examples/abc.rs [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+fn get_semitone(note: Note, accidental: Option<Accidental>, key_signature: i8) -> i32 {
+    let accidental = accidental.unwrap_or_else(|| match note {
+        Note::B if key_signature >= 7 => Accidental::Sharp,
+        Note::E if key_signature >= 6 => Accidental::Sharp,
+        Note::A if key_signature >= 5 => Accidental::Sharp,
+        Note::D if key_signature >= 4 => Accidental::Sharp,
+        Note::G if key_signature >= 3 => Accidental::Sharp,
+        Note::C if key_signature >= 2 => Accidental::Sharp,
+        Note::F if key_signature >= 1 => Accidental::Sharp,
+        Note::B if key_signature <= -1 => Accidental::Flat,
```
Comment (alsuren) [inline messages/examples/abc.rs:170]: could probably add a newtype which derives PartialOrd, if you wanted to avoid passing the primitive u8 around

Comment (qwandor) [reply messages/examples/abc.rs:170]: Could do, but I'm not sure it's really worth it for a type which is only used internally in this example for passing between two functions. If this were part of a serious library I would.

File: messages/examples/abc.rs (hunk 2 above)
Comment (qwandor) [reply messages/examples/abc.rs:44]: Done.

File: messages/examples/abc.rs (hunk 3 above)
Comment (qwandor) [reply messages/examples/abc.rs]: Slowed it down.

File: messages/src/client.rs (hunk 1 above)
Comment (qwandor) [reply messages/src/client.rs:102]: Done.

**Pull request merged**